src/nni_manager/core/nnimanager.ts

```diff
@@ -425,10 +427,12 @@ class NNIManager implements Manager {
                 if (this.customizedTrials.length > 0) {
                     const hyperParams: string | undefined = this.customizedTrials.shift();
                     this.dispatcher.sendCommand(ADD_CUSTOMIZED_TRIAL_JOB, hyperParams);
-                } else {
-                    this.dispatcher.sendCommand(REQUEST_TRIAL_JOBS, '1');
+                    requestTrialNum --;
```

Comment:
why `requestTrialNum --`?

Reply:
I got the reason, but this might be not correct? because requestTrialNum in line 430 is the same as requestTrialNum in line 425.

Reply:
This requestTrialNum-- is used to ensure that custom trials share same concurrent trial number with non custom trials.

Reply:
Updated, logic is same, but looks clearer.